### PR TITLE
feat(core, category): introduce DaffSortable for collections of elements

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/core/state/src/collection/sortable.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  Descending = 'DSC'
+}
+
+/**
+ * Describes the properties of a collection that make the collection able to be
+ * sorted by Daffodil.
+ */
+export interface DaffSortable {
+  sort_options: DaffSortOption[];
+  applied_sort_option: string;
```
Comment (griest024) [inline libs/core/state/src/collection/sortable.ts:25]: I assume this is pulled from `DaffSortOption#label`? If so I would type it as `DaffSortOption['label']`.

**Review by griest024 — CHANGES_REQUESTED** [review]
What's your rationale for putting this in state instead of the root core package?

File: libs/core/state/src/collection/sortable.ts (hunk 1 above)
Comment (damienwebdev) [reply libs/core/state/src/collection/sortable.ts]: e8b7bc54

**Comment by damienwebdev** [comment]
@griest024 I think, but can't say for certain, that almost every single thing here is related to state and state alone. Collections don't really occur in the driver layer.

**Comment by damienwebdev** [comment]
I take what I said back -- I found usages within the driver layers.

File: libs/core/state/src/collection/sortable.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  Descending = 'DSC'
+}
+
+/**
+ * Describes the properties of a collection that make the collection able to be
+ * sorted by Daffodil.
+ */
+export interface DaffSortable {
+  /**
+   * The available options to sort the collection by.
```
Comment (griest024) [inline libs/core/state/src/collection/sortable.ts:25]: *by which* to sort the collection

**Review by griest024 — APPROVED** [review]

File: libs/core/state/src/collection/sortable.ts (hunk 2 above)
Comment (damienwebdev) [reply libs/core/state/src/collection/sortable.ts]: e6dc89d5

Comment (griest024) [reply libs/core/state/src/collection/sortable.ts]: I like `The available options by which to sort the collection` better

**Comment by damienwebdev** [comment]
Needs review by @lderrickable

**Review by lderrickable — APPROVED** [review]

**Pull request merged**